### PR TITLE
fix(dereference): modification for circular ref for delete objects

### PR DESCRIPTION
**Pull request by SB-rohitdesai**
STOP-184
## Motivation and Context
Modified the code for faster execution.

## How Has This Been Tested?
Tested locally with npm run test.update

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] This PR's code follows as closely as possible [the coding style/guidelines of this project](???).
- [ ] I have added error reporting and followed [the error reporting guidelines](???).
- [ ] I have added event tracking and followed [the event tracking guidelines](???).
- [ ] I have updated any relevant documentation accordingly to reflect this PR's changes.
- [ ] I have added automated tests (unit/integration/e2e/other) to cover my changes.
- [ ] All new and existing tests pass locally (excluding flaky CI tests).

### CODE REVIEW

**Review by prafullaAtSB — APPROVED** [review]
changes are good to go ahead

File: src/decycle.ts [hunk 1]
```diff
@@ -1,36 +1,46 @@
...
-export const decycle = (obj: unknown, replacer?: (value: any) => any) => {
+export function decycle(obj: unknown, replacer?: (value: any) => any) {
   const objs = new WeakMap<object, string>();
-  return (function derez(value: any, path: string[]) {
-    // The new object or array
-    let curObj: any;
-
-    // If a replacer function was provided, then call it to get a replacement value.
-    if (replacer) value = replacer(value);
+  const objectsToDelete: object[] = []; // To keep track of objects to delete later
```
Comment (bhaskarsontakke) [inline src/decycle.ts:6]: rename objectsToDelete to objectsTobeDeleted

Comment (SB-rohitdesai) [reply src/decycle.ts]: changes are done

**Review by bhaskarsontakke — APPROVED** [review]

**Review by bhaskarsontakke — APPROVED** [review]

**Review by prafullaAtSB — APPROVED** [review]
Good to go ahead

**Comment by Didel** [comment]
It looks like these changes are the root cause of #146 , which will surface as soon as people update their @stoplight/prism-cli dependency. Could one of you have a look at #146 and test that with the changes from this PR?

**Pull request merged**